### PR TITLE
Add animation to navbar

### PR DESCRIPTION
**Pull request by xinyee20**
Navbar will hide when scrolling down and appear and scrolling up.
Position is fixed so that navbar can appear when scrolling up and is not along the same plane as the page content.

https://user-images.githubusercontent.com/63994179/149696509-d9567527-bd27-4d20-b0b3-df28a1c75e81.mp4

Additional edits:
Adjust the way the pageName is obtained.

### CODE REVIEW

**Comment by xinyee20** [comment]
[Animation for Navigation bar](https://trello.com/c/SypmzZ4W/6-animation-for-navigation-bar)

**Comment by xinyee20** [comment]
[Animation for Navigation bar](https://trello.com/c/SypmzZ4W/6-animation-for-navigation-bar)

**Comment by xinyee20** [comment]
[Animation for Navigation bar](https://trello.com/c/SypmzZ4W/6-animation-for-navigation-bar)

File: src/components/NavBar.tsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import { useNavigate } from 'react-router-dom'
-import { MainContainer, NavContainer } from './styles/NavBar.styled'
+import { useEffect, useState } from 'react'
```
Comment (hui444) [inline src/components/NavBar.tsx:2]: move this to line 1?

File: src/components/NavBar.tsx [hunk 2]
```diff
@@ -40,9 +41,27 @@ function NavBar() {
...
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [prevScrollPos, visible, handleScroll])
+
   return (
     <NavContainer>
-      <MainContainer>{renderedNavItems}</MainContainer>
+      <MainContainer>
+        <div style={{ ...navBarStyles, top: visible ? '0' : '-60px', position: 'fixed' }}>{renderedNavItems}</div>
```
Comment (hui444) [inline src/components/NavBar.tsx:62]: consider use styled components instead of inline styling

File: src/components/styles/NavBar.styled.ts [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
 export const NavContainer = styled.div`
   width: 100vw;
   background: ${(props) => props.theme.palette.common.black};
 `
+
+export const BufferContainer = styled.div`
+  height: 30px;
+`
+
+export const navBarStyles = {
```
Comment (hui444) [inline src/components/styles/NavBar.styled.ts:16]: consider using styled components

**Review by hui444 — APPROVED** [review]
lgtm! try and use styled components instead of inline styling!

File: src/components/NavBar.tsx (hunk 1 above)
Comment (xinyee20) [reply src/components/NavBar.tsx:1]: 5edc1cc

File: src/components/NavBar.tsx (hunk 2 above)
Comment (xinyee20) [reply src/components/NavBar.tsx]: resolved in 5edc1cc

File: src/components/styles/NavBar.styled.ts (hunk 3 above)
Comment (xinyee20) [reply src/components/styles/NavBar.styled.ts]: resolved in 5edc1cc

File: src/components/NavBar.tsx [hunk 4]
```diff
@@ -40,9 +41,25 @@ function NavBar() {
 
   const renderedNavItems = navTitles.map((item) => renderNavItem(item))
 
+  const [prevScrollPos, setPrevScrollPos] = useState(0)
+  const [visible, setVisible] = useState(true)
```
Comment (hui444) [inline src/components/NavBar.tsx:45]: maybe isVisible would be better since it's a boolean

**Review by hui444 — APPROVED** [review]
minor naming issue but otherwise lgtm!

File: src/components/NavBar.tsx (hunk 4 above)
Comment (xinyee20) [reply src/components/NavBar.tsx]: resolved in 00cc5bc

**Pull request merged**